src/Renci.SshNet/Common/Pipe.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Renci.SshNet.Common
+{
+    using System;
+    using System.IO;
+
+    /// <summary>
+    /// A generic pipe to pass through data.
+    /// </summary>
+    internal class Pipe : IDisposable
+    {
```

Comment:
Personally, I prefer to remove both properties. However, I want to limit the number of breaking changes.  We could start by marking them as obsolete in the next release (after 2016.1.0).

For now, I guess both properties must be able to return the incomplete output (up until that moment in time).

This means we cannot use ```StreamReader.ReadToEnd()``` as that would block until the command has finished (the writer has completed).

Guess we'll have to do something like this:

```cs
var buffer = new byte[4096];
while (_outputReadStream.DataAvailable)
{
    var bytesRead = _outputReadStream.Read(buffer, 0, buffer.Length);
    _result.Append(_encoding.GetString(buffer, 0, bytesRead));
}
```